### ISSUE TITLE
Match against interval fails

### ISSUE THREAD

**Issue by mindbound** [kmsquire/Match.jl#26]
![Screenshot](https://i.imgur.com/x7pHdR5.png)

**Comment by TotalVerb** [comment]
What if you actually wanted to match for `1:10`? I don't think making intervals or collections use `in` by default is a good idea. Racket doesn't do this:

```
> (match 5    
    ['(1 2 3 4 5 6 7 8 9 10) "matched"]
    [_                       "not matched"])
"not matched"
```

It would be nice to have some syntax, like say, `@in(1:10)`, that does this.

**Comment by TotalVerb** [comment]
Or simply expand `x in c` to `(x, if x in @val(c))`, where `@val` is a hypothetical macro that protects the inner expression from being interpreted as a match condition. This does require that `if` work inside a more complicated match expression, which at present it doesn't seem like it does.

**Comment by kmsquire** [comment]
In fact, [taking inspiration from Rust](https://doc.rust-lang.org/book/patterns.html#ranges), this is actually supported:

```julia
julia> f(x) = @match x begin
          1:10 => "x"
          _    => "???"
       end

julia> f(10)
"x"

julia> f(11)
"???"

julia> f(1:10)  # explicitly matching the range also works
"x"

julia> f(1:5)    # if the input is a range, it must match exactly
"???"
```

It could be argued that this is a little magical, but it is pretty useful.  (Rust actually uses `..` for creating ranges, and `...` for matching against them.)

I'm actually confused by the problem @mindbound had, since it works fine for me:

```julia
julia> x = 5
5

julia> @match x begin
          (1:10) => println("$x")
          _ => println("???")
       end
5

julia> VERSION
v"0.4.3-pre+6"
```

@mindbound, what version of `Match` do you have installed?

**Comment by kmsquire** [comment]
Whoops... I just realized that I hadn't tagged a version after adding that feature.   I just did so--v0.4.1 should be available in a little while, which allows matching against ranges.

**Comment by kmsquire** [comment]
Sorry, I meant that `v0.2.0` would be available in a little while.

**Comment by kmsquire** [comment]
Should be available now, if you're running v0.4 and do a `Pkg.update()`

**Comment by mindbound** [comment]
Works now, thanks.
